### PR TITLE
[improvement](catalog) Change the push-down parameters of the predicate function of the table query SQL into variables

### PR DESCRIPTION
**Pull request by zy-kkk**
## Proposed changes

Issue Number: close #xxx

In this PR, we will control whether the external data source query is a push-down function parameter in the filter condition, changing the enable_fun_pushdown of fe conf to the enable_ext_func_pred_pushdown of the variable

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/jdbc/JdbcScanNode.java [hunk 1]
```diff
@@ -313,7 +312,7 @@ private static boolean shouldPushDownConjunct(TOdbcTableType tableType, Expr exp
         if (containsFunctionCallExpr(expr)) {
             if (tableType.equals(TOdbcTableType.MYSQL) || tableType.equals(TOdbcTableType.CLICKHOUSE)
                     || tableType.equals(TOdbcTableType.ORACLE)) {
-                return Config.enable_func_pushdown;
+                return ConnectContext.get().getSessionVariable().enableExtFuncPredPushdown;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/jdbc/JdbcScanNode.java:315]: ConnectContext.get() may be null

File: fe/fe-core/src/main/java/org/apache/doris/planner/external/odbc/OdbcScanNode.java [hunk 2]
```diff
@@ -267,6 +266,6 @@ public static boolean shouldPushDownConjunct(TOdbcTableType tableType, Expr expr
                 return false;
             }
         }
-        return Config.enable_func_pushdown;
+        return ConnectContext.get().getSessionVariable().enableExtFuncPredPushdown;
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/planner/external/odbc/OdbcScanNode.java:269]: ditto

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 3]
```diff
@@ -522,7 +524,11 @@ public class SessionVariable implements Serializable, Writable {
     @VariableMgr.VarAttr(name = EXPAND_RUNTIME_FILTER_BY_INNER_JION)
     public boolean expandRuntimeFilterByInnerJoin = true;
 
-    @VariableMgr.VarAttr(name = JDBC_CLICKHOUSE_QUERY_FINAL)
+    /**
```
Comment (morningman) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:527]: No need to commend, add `description` field of `@VariableMgr.VarAttr`

**Review by morningman — APPROVED** [review]
LGTM

**Comment by zy-kkk** [comment]
run buildall

**Review by LemonLiTree — APPROVED** [review]

**Pull request merged**